subsys/net/ip/net_context.c

```diff
@@ -1544,6 +1549,86 @@ static int get_context_dscp_ecn(struct net_context *context,
 #endif
 }
 
+static int get_context_ttl(struct net_context *context,
+				 void *value, size_t *len)
+{
+#if defined(CONFIG_NET_IPV4)
+	*((int *)value) = context->ipv4_ttl;
+
+	if (len) {
+		*len = sizeof(int);
+	}
+
+	return 0;
+#else
```

Comment:
Did you not introduce generic getter/setter function to be used here?

Reply:
For the ttl and hoplimit getters I did not, the `get_uint8_option()` expects user to supply uint8_t value but the value for the ttl and hoplimit is typically int, so the values were incompatible. Then I thought about creating `get_int_option()` but because we are storing ttl and hoplimit as uint8_t the function was not that nice looking. I will investigate this option getter/setter dilemma in separate PR that I mentioned in the refactoring patch.